server/plugin.go

```diff
@@ -328,68 +308,58 @@ func (p *Plugin) handleRegisterMeetingFromOnlineVersion(w http.ResponseWriter, r
+			appErr.StatusCode
 	}
 
 	if appErr = p.API.KVSet(fmt.Sprintf("%v%v", PostMeetingKey, req.MeetingID), []byte(post.Id)); appErr != nil {
-		fmt.Println(appErr.Error())
-		http.Error(w, appErr.Error(), appErr.StatusCode)
-		return
+		return errors.Wrapf(appErr, "cannot register meeting. An error occured while saving the meeting ID in the database"),
+			appErr.StatusCode
 	}
 
 	w.Write([]byte(fmt.Sprintf("%v", req.MeetingID)))
+
+	return nil, http.StatusOK
 }
 
-func (p *Plugin) handleCreateMeetingInServerVersion(w http.ResponseWriter, r *http.Request) {
+func (p *Plugin) handleCreateMeetingInServerVersion(w http.ResponseWriter, r *http.Request) (error, int) {
 	config := p.getConfiguration()
 	if config.ProductType == productTypeOnline {
-		mlog.Error("Cannot create meeting in the server version when the online is set")
-		http.Error(w, "Forbidden", http.StatusForbidden)
-		return
+		return errors.New("Cannot create meeting. Product Type is not set as 'server'"), http.StatusForbidden
 	}
 
 	userID := r.Header.Get("Mattermost-User-Id")
 	if userID == "" {
-		mlog.Error("Request doesn't have Mattermost-User-Id header")
-		http.Error(w, "Not authorized", http.StatusUnauthorized)
-		return
+		return errors.New("Cannot create meeting. Missing 'Mattermost-User-Id' header"), http.StatusUnauthorized
 	}
 
 	var user *model.User
 	var appError *model.AppError
 	user, appError = p.API.GetUser(userID)
 	if appError != nil {
-		mlog.Error("Error getting user: " + appError.Error())
-		http.Error(w, appError.Error(), appError.StatusCode)
-		return
+		return errors.Wrapf(appError, "cannot create meeting. An error occured while fetching user"),
+			appError.StatusCode
 	}
 
 	if user == nil {
-		mlog.Error("User with that id doesn't exist: " + userID)
-		http.Error(w, "User is nil", http.StatusUnauthorized)
-		return
+		return fmt.Errorf("cannot create meeting. User with that id doesn't exist: %s", userID),
+			http.StatusUnauthorized
 	}
 
 	var req StartServerMeetingRequest
 	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
-		mlog.Error("Error decoding JSON body: " + err.Error())
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
+		return errors.Wrapf(err, "cannot create meeting. An error occured while decoding JSON body"), http.StatusBadRequest
 	}
 
 	if _, err := p.API.GetChannelMember(req.ChannelID, user.Id); err != nil {
-		mlog.Error("Error getting channel member: " + err.Error())
-		http.Error(w, "Forbidden", http.StatusForbidden)
-		return
+		return errors.Wrapf(err, "cannot create meeting. An error occured while fetching channel membership"),
+			http.StatusForbidden
 	}
 
 	applicationState, apiErr := p.fetchOnlineMeetingsURL()
 	if apiErr != nil {
-		mlog.Error("Error fetching meetings resource url: " + apiErr.Message)
-		http.Error(w, apiErr.Message, http.StatusInternalServerError)
-		return
+		return fmt.Errorf("cannot create meeting. An error occured while fetching meetings resource URL: %+v", apiErr),
```

Comment:
You can use errors.Wrap here

Reply:
I can't because `apiErr` variable doesn't implement the `error` interface.